tests/test_generators/test_pydanticgen.py

```diff
@@ -1,17 +1,46 @@
 import pytest
 import yaml
+from click.testing import CliRunner
 from linkml_runtime import SchemaView
 from linkml_runtime.dumpers import yaml_dumper
 from linkml_runtime.linkml_model import SlotDefinition
 from linkml_runtime.utils.compile_python import compile_python
 from pydantic import ValidationError
 
+from linkml.generators import pydanticgen
 from linkml.generators.pydanticgen import PydanticGenerator
 from linkml.utils.schema_builder import SchemaBuilder
 
 PACKAGE = "kitchen_sink"
 
 
+@pytest.fixture
+def runner():
+    return CliRunner()
+
+
+@pytest.mark.parametrize(
+    "arguments",
+    [
+        ["--pydantic-version", "1"],
+        ["--pydantic-version", "2"],
+        ["--pydantic_version", "1"],
+        ["--pydantic_version", "2"],
+        ["--extra-fields", "allow"],
+    ],
+)
+def test_metamodel_valid_cli_arguments(arguments, kitchen_sink_path):
+    runner = CliRunner()
+    arguments_str = " ".join(arguments)
+    result = runner.invoke(pydanticgen.cli, f"{arguments_str} {kitchen_sink_path}")
```

Comment:
Unfortunately I've learned that passing a string as the second argument to `runner.invoke` doesn't work on Windows (because that signature relies on the `shlex` [module](https://docs.python.org/3/library/shlex.html) under the hood). Gotta use a list of strings instead.